samples/updatehub/README.rst

```diff
@@ -0,0 +1,246 @@
+  MCUboot can be found in :ref:`mcuboot`.
+
+Building and Running
+********************
+
+The below steps describe how to build and run the ``UpdateHub`` sample in
+Zephyr. Where examples are given, it is assumed the sample is being built for
+the Freedom-K64F Development Kit (``BOARD=fdrm_k64f``).
+
+Step 1: Build MCUboot
+=====================
+
+Build MCUboot by following the instructions in the :ref:`mcuboot` documentation
+page.
+
+Step 2: Flash MCUboot
+=====================
+
+Flash the resulting image file to the 0x0 address of the flash memory. This can
+be done in multiple ways, but the most common ones would be using make or ninja:
+
+.. code-block:: console
+
+   make flash
+   # or
+   ninja flash
+
+Step 3: Build UpdateHub
+==============================
+
+``UpdateHub`` can be built for the fdrm_k64f as follows:
+
+.. zephyr-app-commands::
+    :zephyr-app: samples/updatehub/updatehub_<manual-or-polling>
+    :board: fdrm_k64f
+    :build-dir: fdrm_k64f_build
+    :goals: build
+
+.. updatehub_sample_sign:
+
+Step 4: Sign the first image
+============================
+
+From this section onwards you can use either a binary (``.bin``) or an Intel
+Hex (``.hex``) image format. This is written as ``(bin|hex)`` in this
+document, so edit the commands accordingly before running them.
+
+Using MCUboot's :file:`imgtool.py` script, sign the :file:`zephyr.(bin|hex)`
+file you built in Step 3. In the below example, the MCUboot repo is located at
+:file:`~/src/mcuboot`.
+
+.. code-block:: console
+
+   ~/src/mcuboot/scripts/imgtool.py sign \
+        --key ~/src/mcuboot/root-rsa-2048.pem \
+        --header-size 0x200 \
+        --align 8 \
+        --version 1.0.0 \
+        --slot-size <image-slot-size> \
+        <path-to-zephyr.(bin|hex)> signed.(bin|hex)
+
+The command above creates an image file called :file:`signed.(bin|hex)` in the
+current directory.
+
+Step 5: Flash the first image
+=============================
+
+Upload the :file:`signed.(bin|hex)` file from Step 4 to image slot-0 of your
+board.  The location of image slot-0 varies by board, as described in
+:ref:`mcuboot_partitions`.  For the fdrm_k64f, slot-0 is located at address
+``0xc000``.
+
+Using :file:`pyocd` you don't need to specify the slot-0 starting address.
+
+.. code-block:: console
+
+    sudo pyocd-flashtool <path-to-signed.bin>
+
+
+Step 6: Signing the test image
+================================
+
+For the update to be correctly validated on the server you will need sign the
+(``bin|hex``) image, piping the output to another file.
+
+.. code-block:: console
+
+   ~/src/mcuboot/scripts/imgtool.py sign \
+        --key ~/src/mcuboot/root-rsa-2048.pem \
+        --header-size 0x200 \
+        --align 8 \
+        --version 2.0.0 \
+        --slot-size <image-slot-size> \
+        <path-to-zephyr.(bin|hex)> signed_v2.(bin|hex)
+
+
+Step 7: Create a package with UpdateHub Utilities (``uhu``)
+=======================================================
```

Comment:
The title underline must be at least as long as the title above it, so add a few more ``===``'s